### PR TITLE
refactor: use new DateFormatter for CSVExportButton

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Closes https://github.com/influxdata/ui/issues/1906

Replaces the `moment` library with the new Date-Formatter in `CSVExportButton`. This PR also replaces the filename spaces with `_`.

Notes for the reviewer: 
* ~The new formatter output is slightly different format, as seen in the pictures. In both cases, `YYYY-MM-DD-HH-mm` was requested.~ [ just learned `YYYY-MM-DD HH:mm` is the correct replacement for `YYYY-MM-DD-HH-mm` ]

Before
Screen Shot 2021-07-06 at 8 48 07 AM

After
Screen Shot 2021-07-06 at 11 04 10 AM

### CODE REVIEW

File: src/shared/components/CSVExportButton.tsx [hunk 1]
```diff
@@ -52,8 +52,9 @@ class CSVExportButton extends PureComponent<StateProps, {}> {
 
   private handleClick = () => {
     const {files} = this.props
+    const formatter = createDateTimeFormatter('YYYY-MM-DD-HH-mm')
```
Comment (hoorayimhelping) [inline src/shared/components/CSVExportButton.tsx:55]:
```suggestion
    const formatter = createDateTimeFormatter('YYYY-MM-DD HH:mm')
```

couple of changes to make this a proper format

**Review by hoorayimhelping — CHANGES_REQUESTED** [review]

File: src/shared/components/CSVExportButton.tsx (hunk 1 above)
Comment (ChitlangeSahas) [reply src/shared/components/CSVExportButton.tsx]: Just made the change! This makes more sense!

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 61 | :yellow_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=master)

File: src/shared/components/CSVExportButton.tsx [hunk 2]
```diff
@@ -52,8 +52,10 @@ class CSVExportButton extends PureComponent<StateProps, {}> {
 
   private handleClick = () => {
     const {files} = this.props
+    const formatter = createDateTimeFormatter('YYYY-MM-DD HH:mm')
     const csv = files.join('\n\n')
-    const now = moment().format('YYYY-MM-DD-HH-mm')
+    let now = formatter.format(new Date())
```
Comment (hoorayimhelping) [inline src/shared/components/CSVExportButton.tsx:57]: it's probably slightly better to declare `now` as a `const` and replace the spaces when we initialize it:
```ts
formatter.format(new Date()).replace(/\s+/gi, '_')
```
but this is fine for now

**Review by hoorayimhelping — APPROVED** [review]

**Pull request merged**